packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -162,4 +172,29 @@ export class SelectedNetworkController extends BaseControllerV2<
     }
     return this.state.domains[METAMASK_DOMAIN];
   }
+
+  /**
+   * Accesses the provider and block tracker for the currently selected network.
+   *
+   * @param domain - the domain for the provider
+   * @returns The proxy and block tracker proxies.
+   */
+  getProviderAndBlockTracker(domain: Domain): NetworkProxy {
```

Comment:
hmm yeah I copied the name/code from the netwokcontroller where it creates the proxies. Changing the name to getNetworkClient would be a little confusing since its not the same thing as a NetworkController's concept of a NetworkClient (its missing configuration and destroy). ATM we only need these proxies to make the middleware chain work without major modification, so theres really no need to put proxy to configuration and destroy into the mix. 

So yeah, I think getNetworkClient would be a confusing name given that it doesnt return a NetworkClient, and does not re-use that name to mean something different in the context of queued request controller.